exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdDistinctRowCount.java

```diff
@@ -229,4 +321,42 @@ public Double getDistinctRowCount(DrillJoinRelBase joinRel, RelMetadataQuery mq,
     }
     return RelMdUtil.numDistinctVals(distRowCount, mq.getRowCount(joinRel));
   }
+
+  private ImmutableBitSet getSingleGbyKey(ImmutableBitSet groupKey, int idx) {
+    if (groupKey.get(idx)) {
+      return ImmutableBitSet.builder().set(idx, idx+1).build();
+    } else {
+      return null;
+    }
+  }
+
+  private double getPredSelectivityContainingInputRef(RexNode predicate, int inputRef,
+      RelMetadataQuery mq, TableScan scan) {
+    if (predicate instanceof RexCall) {
+      if (predicate.getKind() == SqlKind.AND) {
+        double sel, andSel = 1.0;
+        for (RexNode op : ((RexCall) predicate).getOperands()) {
+          sel = getPredSelectivityContainingInputRef(op, inputRef, mq, scan);
+          andSel *= sel > 0 ? sel : 1.0;
```

Comment:
`sel > 0` already includes the value 1.0 .. can you clarify this setting ?

Reply:
Simplified it to avoid confusion